File: C#Integration/PokerProtocol/Commands/Lobby/JoinTableCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using PokerWorld.Game;

namespace PokerProtocol.Commands.Lobby
{
    public class JoinTableCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return JoinTableCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyJOIN_TABLE";

        private readonly string m_TableName;
        private readonly string m_PlayerName;

        public string TableName
        {
            get { return m_TableName; }
        } 

        public string PlayerName
        {
            get { return m_PlayerName; }
        } 

        public JoinTableCommand(StringTokenizer argsToken)
        {
            m_TableName = argsToken.NextToken();
            m_PlayerName = argsToken.NextToken();
        }

        public JoinTableCommand(string p_tableName, int p_bigBlind, int p_maxPlayers, string p_playerName, int wtaPlayerAction, int wtaBoardDealed, int wtaPotWon, TypeBet limit)
        {
            m_TableName = p_tableName;
            m_PlayerName = p_playerName;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_TableName);
            Append(sb, m_PlayerName);
        }

        public string EncodeResponse( int seat )
        {
            return new JoinTableResponse(this, seat).Encode();
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/protocol/commands/lobby/JoinTableCommand.java
package protocol.commands.lobby;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;
import protocol.commands.lobby.response.JoinTableResponse;

public class JoinTableCommand implements ICommand
{
    private final String m_tableName;
    private final String m_playerName;
    public static String COMMAND_NAME = "lobbyJOIN_TABLE";
    
    public JoinTableCommand(StringTokenizer argsToken)
    {
        m_playerName = argsToken.nextToken();
        m_tableName = argsToken.nextToken();
    }
    
    public JoinTableCommand(String pName, String tName)
    {
        m_playerName = pName;
        m_tableName = tName;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(JoinTableCommand.COMMAND_NAME);
        sb.append(Command.DELIMITER);
        sb.append(m_playerName);
        sb.append(Command.DELIMITER);
        sb.append(m_tableName);
        sb.append(Command.DELIMITER);
        return sb.toString();
    }
    
    public String encodeResponse(int noSeat)
    {
        return new JoinTableResponse(m_playerName, m_tableName, noSeat).encodeCommand();
    }
    
    public String encodeErrorResponse()
    {
        return new JoinTableResponse(m_playerName, m_tableName, -1).encodeCommand();
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
    
    public String getTableName()
    {
        return m_tableName;
    }
}
